pkg/debug/checkups/services_windows.go

```diff
@@ -0,0 +1,285 @@
+		return "SERVICE_WIN32_SHARE_PROCESS"
+	case 16 | 32:
+		return "SERVICE_WIN32"
+	case 256:
+		return "SERVICE_INTERACTIVE_PROCESS"
+	case 1 | 2 | 8:
+		return "SERVICE_DRIVER"
+	case 1 | 2 | 4 | 8 | 16 | 32 | 256:
+		return "SERVICE_TYPE_ALL"
+	default:
+		return fmt.Sprintf("unknown: %d", serviceType)
+	}
+}
+
+func startTypeHumanReadable(startType uint32) string {
+	// For mapping, see https://pkg.go.dev/golang.org/x/sys/windows/svc/mgr#pkg-constants
+	// and https://pkg.go.dev/golang.org/x/sys@v0.11.0/windows#SERVICE_BOOT_START
+	switch startType {
+	case 0:
+		return "SERVICE_BOOT_START"
+	case 1:
+		return "SERVICE_SYSTEM_START"
+	case mgr.StartAutomatic:
+		return "SERVICE_AUTO_START"
+	case mgr.StartManual:
+		return "SERVICE_DEMAND_START"
+	case mgr.StartDisabled:
+		return "SERVICE_DISABLED"
+	default:
+		return fmt.Sprintf("unknown: %d", startType)
+	}
+}
+
+func errorControlHumanReadable(errorControl uint32) string {
+	// For mapping, see https://pkg.go.dev/golang.org/x/sys/windows/svc/mgr#pkg-constants
+	switch errorControl {
+	case mgr.ErrorIgnore:
+		return "SERVICE_ERROR_IGNORE"
+	case mgr.ErrorNormal:
+		return "SERVICE_ERROR_NORMAL"
+	case mgr.ErrorSevere:
+		return "SERVICE_ERROR_SEVERE"
+	case mgr.ErrorCritical:
+		return "SERVICE_ERROR_CRITICAL"
+	default:
+		return fmt.Sprintf("unknown: %d", errorControl)
+	}
+}
+
+func sidTypeHumanReadable(sidType uint32) string {
+	// For mapping, see https://pkg.go.dev/golang.org/x/sys@v0.11.0/windows#SERVICE_SID_TYPE_NONE
+	// See also: https://learn.microsoft.com/en-us/windows/win32/api/winsvc/ns-winsvc-service_sid_info
+	switch sidType {
+	case 0:
+		return "SERVICE_SID_TYPE_NONE"
+	case 1:
+		return "SERVICE_SID_TYPE_UNRESTRICTED"
+	case 3:
+		return "SERVICE_SID_TYPE_RESTRICTED"
+	default:
+		return fmt.Sprintf("unknown: %d", sidType)
+	}
+}
+
+func gatherServices(z *zip.Writer, serviceManager *mgr.Mgr) error {
+	services, err := serviceManager.ListServices()
+	if err != nil {
+		return fmt.Errorf("listing services: %w", err)
+	}
+
+	servicesOut, err := z.Create("services.json")
+	if err != nil {
+		return fmt.Errorf("creating services.json: %w", err)
+	}
+
+	servicesRaw, err := json.Marshal(services)
+	if err != nil {
+		return fmt.Errorf("marshalling services: %w", err)
+	}
+
+	if _, err := servicesOut.Write(servicesRaw); err != nil {
+		return fmt.Errorf("writing services: %w", err)
+	}
+
+	return nil
+}
+
+func gatherServiceManagerEventLogs(ctx context.Context, z *zip.Writer) error {
+	cmdletArgs := []string{
+		"Get-EventLog",
+		"-Newest", "50",
+		"-LogName", "System",
+		"-Source", "\"Service Control Manager\"",
+		"-Message", fmt.Sprintf("*%s*", kolideSvcName),
+		"|",
+		"Format-Table", "-Wrap", "-AutoSize", // ensure output doesn't get truncated
+	}
+
+	getEventLogCmd := exec.CommandContext(ctx, "powershell.exe", cmdletArgs...)
+	getEventLogCmd.SysProcAttr = &syscall.SysProcAttr{HideWindow: true} // prevents spawning window
+	out, err := getEventLogCmd.CombinedOutput()
```

Comment:
You might be able to create the writer, and then set `getEventLogCmd.Stdout` and getEventLogCmd.Stderr` to the writer directly.

Reply:
Yup, that works! Updated